sycl/test-e2e/Basic/large-range.cpp

```diff
@@ -26,34 +26,42 @@ void check_sum(std::string_view desc, const ContainerT &data, size_t N) {
 template <typename RangeT>
 void test_regular(std::string_view desc, queue &q, size_t B, RangeT range) {
   auto N = range.size();
-  std::vector accumulators_v(B, 0, usm_allocator<int, usm::alloc::shared>(q));
-  auto *accumulators = accumulators_v.data();
-
-  q.parallel_for(range, [=](auto it) {
-     atomic_ref<int, memory_order::relaxed, memory_scope::device> ref(
-         accumulators[it.get_linear_id() % B]);
-     ++ref;
-   }).wait();
-
+  std::vector accumulators_v(B, 0);
+  sycl::buffer accumulator_buf{accumulators_v};
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor accumulators{accumulator_buf, h};
+    h.parallel_for(range, [=](auto it) {
+      atomic_ref<int, memory_order::relaxed, memory_scope::device> ref(
+          accumulators[it.get_linear_id() % B]);
+      ++ref;
+    });
+  });
+  sycl::host_accessor host_acc{accumulator_buf};
+  for (int i = 0; i < accumulators_v.size(); ++i) {
+    accumulators_v[i] = host_acc[i];
+  }
```

Comment:
```suggestion
  std::vector accumulators_v(B, 0);
  {
    sycl::buffer accumulator_buf{accumulators_v};
    q.submit([&](sycl::handler &h) {
      sycl::accessor accumulators{accumulator_buf, h};
      h.parallel_for(range, [=](auto it) {
        atomic_ref<int, memory_order::relaxed, memory_scope::device> ref(
            accumulators[it.get_linear_id() % B]);
        ++ref;
       });
     });
   }
```

Buffer automatically copies the data back to the host origin on destruction. The same change can be applied below for `test_spec_constant`

Reply:
Yeah, I was trying to play it safe because I did not want to add a bug in the test case so I went for redundancy but this is good to know, thanks for the information, will update it now!